src/backend/distributed/executor/citus_custom_scan.c

```diff
@@ -801,3 +823,45 @@ IsCitusCustomScan(Plan *plan)
 
 	return true;
 }
+
+
+/*
+ * In a Job, given a list of relations, if all them belong to the same
+ * colocation group, the Job's colocation ID is set to the group ID, else,
+ * it will be set to INVALID_COLOCATION_ID.
+ */
+static void
+SetJobColocationId(Job *job)
+{
+	uint32 jobColocationId = INVALID_COLOCATION_ID;
```

Comment:
We probably need a special case for queries that do not have a distributed table (e.g. outer queries that only query CTEs).

Reply:
Added the exact test case.